x/evm/genesis.go

```diff
@@ -53,6 +58,25 @@ func InitGenesis(
 		panic(fmt.Errorf("error setting params %s", err))
 	}
 
+	txConfig := statedb.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash().Bytes()))
+	stateDB := statedb.New(ctx, k, txConfig)
+
+	for _, hexAddr := range data.Params.EnabledPrecompiles {
+		addr := common.HexToAddress(hexAddr)
+
+		// Set the nonce of the precompile's address (as is done when a contract is created) to ensure
+		// that it is marked as non-empty and will not be cleaned up when the statedb is finalized.
+		stateDB.SetNonce(addr, PrecompileNonce)
+		// Set the code of the precompile's address to a non-zero length byte slice to ensure that the precompile
+		// can be called from within Solidity contracts. Solidity adds a check before invoking a contract to ensure
+		// that it does not attempt to invoke a non-existent contract.
+		stateDB.SetCode(addr, PrecompileCode)
```

Comment:
We could use `Keeper.SetAccount()` for nonce+codehash and `Keeper.SetCode` directly instead of using StateDB since we don't need the snapshot/rollback/commit features of StateDB.

`SetCode` also is global mapping of codeHash to code too, so we only need to set the code once (still need to set codehash in account)

Reply:
This task was split into 2 tasks:
- Init precompiles in `InitGenesis` - this part is postponed, Nick finalizes the spec
- Init precompiles in `UpdateParams` - this part is implemented here: https://github.com/Kava-Labs/ethermint/pull/58

This PR will be closed, sorry for confusion.

Reply:
ah no worries

Reply:
and yeah, in https://github.com/Kava-Labs/ethermint/pull/58 I'm directly using `evmKeeper` instead of `StateDB`